### PR TITLE
Combine masked_copymat and copy_mat

### PR DESCRIPTION
**Pull request by cgmb**
I observed no significant performance differences between the existing and proposed solution in testing.

I also compared the negation via ternary and xor. It made no difference to performance, but the xor check was 3 fewer instructions. So, I figured might as well.

### CODE REVIEW

**Comment by cgmb** [comment]
BTW, if you or @jzuniga-amd need more time to consider this approach, I'm happy enough closing this PR and retargeting it for the next release. This change is entirely about the source code. There's no significant changes to the binary, so it doesn't matter whether it makes the release or not.

File: library/src/include/lib_device_helpers.hpp [hunk 1]
```diff
@@ -94,19 +94,56 @@ enum copymat_direction
...
+ROCSOLVER_KERNEL void copy_mat(copymat_direction direction,
                                      const rocblas_int m,
                                      const rocblas_int n,
                                      U A,
                                      const rocblas_int shiftA,
                                      const rocblas_int lda,
                                      const rocblas_stride strideA,
                                      T* buffer,
-                                     const rocblas_int* mask,
+                                     const Mask mask = no_mask{},
```
Comment (tfalders) [inline library/src/include/lib_device_helpers.hpp:144]: I don't see where you've defined a Mask type. How does that work?

**Review by tfalders — APPROVED** [review]
LGTM! The performance numbers look good. Let me know when you're ready to merge.

File: library/src/include/lib_device_helpers.hpp (hunk 1 above)
Comment (cgmb) [reply library/src/include/lib_device_helpers.hpp:144]: `Mask` is template type, like `T`. The concrete type would be `no_mask`, `info_mask`, or `rocblas_int*`. (I always wrapped things in `info_mask` in this PR to more closely match the existing code, but some of those probably could be left as `rocblas_int*`.)

Comment (tfalders) [reply library/src/include/lib_device_helpers.hpp:144]: Ah, I see it now. Thanks!

Comment (cgmb) [reply library/src/include/lib_device_helpers.hpp:144]: Just to clarify slightly, the fact that this is templated on the Mask type means that the `no_mask` and the `info_mask` versions of the template are separate kernels, directly corresponding to `copy_mat` and `masked_copymat`, respectively.

The template parameter is helpful for deciding how much you want to decide at compile-time vs. run-time. You can push things to compile-time by creating a custom Mask type specifically for that feature, or you can push the logic off until runtime by using a type that takes parameters (like the `info_mask` transform).

**Pull request merged**